sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/DotNetSensorTest.java

```diff
@@ -164,10 +155,64 @@ public void whenReportsArePresentThereAreNoWarnings() {
   }
 
   @Test
-  public void noAnalysisIfNoFilesDetected() {
+  public void whenThereAreBothMainAndTestFiles_doNotLog() {
+    addMainFileToFs();
+    addTestFileToFs();
+    // add roslyn reports to avoid warnings in the logs for this test
+    addRoslynReports();
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.WARN)).isEmpty();
+  }
+
+  @Test
+  public void whenThereAreOnlyTestFiles_logWarning() {
+    addTestFileToFs();
+
+    sensor.execute(tester);
+
+    assertThat(logTester.logs(LoggerLevel.DEBUG)).isEmpty();
+    assertThat(logTester.logs(LoggerLevel.WARN))
+      .containsExactly("This solution contains only TEST files. Because there are no MAIN files, this sensor will be skipped. Your SonarQube/SonarCloud project will not have results for LANG_KEY files.");
```

Comment:
_To consider:_ I'd reorder the given information. First the main action (skip), followed by a reasoning (main/test). And the SQ/SC consequence as is.
```suggestion
      .containsExactly("This sensor will be skipped, because current solution contains only TEST files and no MAIN files. Your SonarQube/SonarCloud project will not have results for LANG_KEY files.");
```